### ISSUE TITLE
How to devide the celeba dataset into trainset and valset?

### ISSUE THREAD

**Issue by CodingMice** [fenglinglwb/MAT#3]
I want to know how  you devide dataset

**Comment by fenglinglwb** [comment]
In the CelebA-HQ dataset, you can find a file 'image_list.txt' to map the image ids to the original CelebA ones. And in the CelebaA dataset, there is a file named 'list_eval_partition' where we just use data labelled with 0 for training and 1 for validation.

**Comment by CodingMice** [comment]
> In the CelebA-HQ dataset, you can find a file 'image_list.txt' to map the image ids to the original CelebA ones. And in the CelebaA dataset, there is a file named 'list_eval_partition' where we just use data labelled with 0 for training and 1 for validation.
Thanks for your reply.  you mean CelebA-HQ dataset(3w) is a subset of CelebaA, and we can use 'list_eval_partition' to mapping the trainval split?

**Comment by fenglinglwb** [comment]
Yes. But at first, you need to map the image ids of CelebA-HQ to those of CelebA.

**Comment by CodingMice** [comment]
> Yes. But at first, you need to map the image ids of CelebA-HQ to those of CelebA.

OK. I've done what you said and got results close to the paper using the pretrained model. BTW, I would like to ask how to generate a mask if I want to test the test set of the celeba-hq

**Comment by fenglinglwb** [comment]
Please refer to 'datasets/mask_generator_512.py' or 'datasets/mask_generator_512_small.py' to generate masks with a 512x512 resolution.

**Comment by klinic** [comment]
> > In the CelebA-HQ dataset, you can find a file 'image_list.txt' to map the image ids to the original CelebA ones. And in the CelebaA dataset, there is a file named 'list_eval_partition' where we just use data labelled with 0 for training and 1 for validation.
> > Thanks for your reply.  you mean CelebA-HQ dataset(3w) is a subset of CelebaA, and we can use 'list_eval_partition' to mapping the trainval split?

Excuse me, You said that CelebA-HQ dataset has 3W pictures? But in paper, the author say that the  CelebA-HQ dataset has 25M pictures. I want to know that how many pictures in CelebA-HQ dataset?

**Comment by lijain** [comment]
> In the CelebA-HQ dataset, you can find a file 'image_list.txt' to map the image ids to the original CelebA ones. And in the CelebaA dataset, there is a file named 'list_eval_partition' where we just use data labelled with 0 for training and 1 for validation

你好，请问这个具体怎么映射？我看到你的masks_val_512_eval是从0开始，这个怎么设置？

**Comment by fenglinglwb** [comment]
for your convenience, I just upload the train/val splits to Google Drive. Find here:
https://drive.google.com/file/d/1ja5Jsc7VYmKbGLXhX_POjeae_jhRjIkT/view?usp=drive_link
https://drive.google.com/file/d/1GB-yxSfdKtDsChTzMwEXDhlDdEzaDG03/view?usp=drive_link

**Comment by lijain** [comment]
感谢你提供的下载地址。不过能请教下这个映射是怎么对应的。我已经处理0,1，2标志的image，分别有24183，2993，2824张图。你的评论中说使用标志1的为val，不过我下载masks_val_512_eval的mask是从0000.png开始的，所以不知道怎么map？

**Comment by lijain** [comment]
> st upload the train/val splits to Google Drive. Find here:

感谢你提供的下载地址。我这边得到了512的3w张图，尝试按照你上述说的'image_list.txt' -->list_eval_partition.txt分类下0,1,2。图片的数量也对，不过我查看到和你给到的val图命名不一致，这个是我没映射对？
![image](https://github.com/fenglinglwb/MAT/assets/18046568/6e9fb1b1-8b5e-4bba-92b1-6e2d14a16d89)

**Comment by fenglinglwb** [comment]
> In the CelebA-HQ dataset, you can find a file 'image_list.txt' to map the image ids to the original CelebA ones.

Hi, you can find the mapping in the 'image_list.txt' file.

**Comment by lijain** [comment]
感谢你的耐心解答。我这边image_list.txt使用这个中ori_file对应的xxx.jpg然后在list_eval_partition.txt中获取0,1,2标志。用你给的val测试fid: 4.8343, pids: 0.1360, uids: 0.2538，但使用我构建的val测试fid: 4.8316, pids: 0.1136, uids: 0.2297和你的结果从差一点。可以详细说下数据集怎么构建？

**Comment by fenglinglwb** [comment]
There exists variance between different machines as randomness is introduced during inference. You may use different random seeds or just take away our released inpainted results directly.

**Comment by lijain** [comment]
我测试用你提供的图片测试结果是基本符合。我用自己那种方式构建val集，测出来的pids和uids差别比较大(两次唯一的差别就是val集不一样，使用的mask等都是你这边提供的)。因此我怀疑是我构建数据集图片不对，想问你具体怎么构建
